src/components/ManaCalculator/calculator.ts

```diff
@@ -0,0 +1,162 @@
+import {
+  GENERATION_PER_SLOT,
+  first_slot_of_epoch,
+  potential_Mana,
+  targetReward,
+  EPOCH_DURATION,
+  decay,
+} from './utils';
+import { ValidatorParameters, ValidatorProps } from './types';
+
+export function calculateManaRewards(
+  stake: number,
+  yourPool: number,
+  validatorParameters: ValidatorParameters,
+  validators: ValidatorProps[],
+  initialEpoch: number,
+  finalEpoch: number,
+  yourRole: 'Validator' | 'Delegator',
+  supply: number,
+): number {
+  let totalTargetReward = 0;
+  const epochDiff = 1;
+  if (finalEpoch) {
+    for (let i = 0; i < epochDiff; i++) {
+      totalTargetReward += decay(
+        targetReward(initialEpoch + i, supply),
+        epochDiff - i,
+      );
+    }
+  } else {
+    finalEpoch = initialEpoch + 1;
+    totalTargetReward = targetReward(initialEpoch, supply);
+  }
```

Comment:
This isn't working properly as you are always calculating initialEpoch + 1

Notice how the `Mana generation per epoch` always stays the same when using `>= 1` as finalEpoch:
![image](https://github.com/iotaledger/iota-wiki/assets/38158676/b8d042ed-3b79-4ce4-942b-e51003e38b54)

![image](https://github.com/iotaledger/iota-wiki/assets/38158676/e95723c6-e26b-40d2-882c-b55e14774587)

Reply:
I did this to allow calculating mana rewards for a single epoch (which is for example needed in the TPS view). In that case no decay is calculated and the epochDiff is `1`.

Reply:
Is this still behaving like that since you pushed https://github.com/iotaledger/iota-wiki/pull/1158#discussion_r1380856710 ? I am now getting different results every time I increase the epoch